os/fs/driver/mtd/mtd_partition.c

```diff
@@ -427,10 +428,9 @@ static int part_ioctl(FAR struct mtd_dev_s *dev, int cmd, unsigned long arg)
 
 	case MTDIOC_BULKERASE: {
 		/* Erase the entire partition */
-
-		ret = priv->parent->erase(priv->parent, priv->firstblock / priv->blkpererase, priv->neraseblocks);
-		if (ret == priv->neraseblocks) {
-			ret = OK;
+		FAR struct mtd_geometry_s *geo = (FAR struct mtd_geometry_s *)arg;
```

Comment:
To prevent erase entire of flash, mtd partition calls erase, instead of bulk erase.

Reply:
You're right. Reverted.